### PR TITLE
Added Attack*.StopToShoot

### PR DESCRIPTION
**Pull request by Mailaender**
~~Closes https://github.com/OpenRA/OpenRA/issues/2327.~~ See http://modenc.renegadeprojects.com/NoMovingFire for reference.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Attack/AttackBase.cs [hunk 1]
```diff
@@ -35,10 +35,13 @@ public abstract class AttackBaseInfo : UpgradableTraitInfo
 
 		[VoiceReference] public readonly string Voice = "Action";
 
+		[Desc("Unable to fire while in motion.")]
+		public readonly bool StopToShoot = false;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Attack/AttackBase.cs]: The name implies that it should actively stop before shooting, but this just prevents it from firing while moving.  This introduces a situation where the unit may not fire at all, which isn't really what #2327 wanted.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Attack/AttackBase.cs]: Renamed it.

**Comment by pchote** [comment]
Note that even with the name change that this doesn't close #2327, which is asking for the targeting/turret behavior of the original units.

**Comment by pchote** [comment]
To be able to solve #2327 this should stay named `StopToShoot`, and also have a check in the turreted attack code to stop moving if the targeted unit is in range.  I don't see a usecase for the feature in its current state, so i'd be hesitant to take it as-is.

**Pull request closed without merging**